core/modules/wpublish/ChunkInventory.cc

```diff
@@ -93,112 +75,67 @@ void fetchDbs(std::string const& instanceName,
+        int         const chunk = std::atoi(((**resultP)[2]).c_str());
+        ChunkInventory::StringSet& ss = chunkMap[chunk];
+        ss.insert(table);
+        nothing = false;
+    }
+    if (nothing)
+        LOGS(_log, LOG_LVL_WARN, "ChunkInventory couldn't find any published chunks for db: " << db);
+}
 
-/// Functor to load db
-class doDb {
-public:
-    doDb(SqlConnection& conn,
-         boost::regex& regex,
-         ChunkInventory::ExistMap& existMap)
-        : _conn(conn),
-          _regex(regex),
-          _existMap(existMap)
-        {}
-
-    void operator()(std::string const& dbName) {
-        // get list of tables
-        std::vector<std::string> tables;
-        SqlErrorObject sqlErrorObject;
-        bool ok = _conn.listTables(tables,  sqlErrorObject, "", dbName);
-        if (!ok) {
-            LOGS(_log, LOG_LVL_ERROR, "SQL error: " << sqlErrorObject.errMsg());
-            assert(ok);
-        }
-        ChunkInventory::ChunkMap& chunkMap = _existMap[dbName];
-        chunkMap.clear(); // Clear out stale entries to avoid mixing.
-        std::for_each(tables.begin(), tables.end(),
-                      doTable(_regex, chunkMap));
-        // All databases get a dummy chunk.
-        // Partitioned databases should already have acceptable dummy chunk
-        // partitioned tables (e.g., Object_1234567890, Source_1234567890)
-        // Non-partitioned databases need a dummy chunk anyway.
-        if (chunkMap.empty()) {
-            // No partitioned tables in this db. Publish an empty chunk anyway.
-            chunkMap[lsst::qserv::DUMMY_CHUNK];
-        } else {
-            // Verify that there is a dummy chunk entry
-            if (chunkMap.find(lsst::qserv::DUMMY_CHUNK) == chunkMap.end()) {
-                LOGS(_log, LOG_LVL_ERROR, "Missing dummy chunk for db=" << dbName);
-
-                // FIXME enable once loader/installer can ensure that the
-                // dummy chunk exists exactly when appropriate
-
-                // std::string msg = "Missing dummy chunk for db=" + dbName;
-                // throw CorruptDbError(msg);
-            }
-        }
-        //std::for_each(chunkMap.begin(), chunkMap.end(), printChunk(std::cout));
-        // TODO: Sanity check: do all tables have the same chunks represented?
-        }
-private:
-    SqlConnection& _conn;
-    boost::regex& _regex;
-    ChunkInventory::ExistMap& _existMap;
-};
+/// Fetch a unique identifier of a worker
+void fetchId(std::string const& instanceName,
+             SqlConnection& sc,
+             std::string& id) {
+
+    // Look for the newest one
+    // FIXME: perhaps we should allow multiple identifiers?
+    std::string const query = "SELECT id,created FROM qservw_" + instanceName + ".Id ORDER BY created DESC LIMIT 1";
+
+    LOGS(_log, LOG_LVL_DEBUG, "Launching query: " << query);
+
+    std::shared_ptr<SqlResultIter> resultP = sc.getQueryIter(query);
+    assert(resultP.get());
```

Comment:
`assert` is not doing anything in production code, and I don't think `getQueryIter` can return `nullptr`

Reply:
Actually, assert is triggered by the unit test when I failed to configure that test (to inject results into MockSql). So, I decided to leave those asserts. They may still make a sense.

Reply:
I look at `getQueryIter` and I still think it cannot return null shared_ptr, so I don't understand how that assert can be triggered.

Reply:
You have to look at the unit test at line 79 of this (my version) of file:
core/modules/wpublish/testChunkInventory.cc

This is where the mock Sql API is constructed:
struct ChunkSql : public MockSql {
...
return std::shared_ptr<SqlIter>();
}

Reply:
Well, I'd argue, that `ChunkSql` is broken as it implements different contract from the base class.

Reply:
`MockSql::getQueryIter()` does not return empty pointer either. I think our code depends on returned pointer to never being zero so I think it should not be allowed to return nullptr.